File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    answer = []
    temp = []
    
    # array 입력값은 배열로 들어온다
    # commands에 2차원 배열로 들어오는데, 2차원은 [i, j, k]
    # i = 인덱스 시작, j = 인덱스 끝, k = 정렬 후 k번때 숫자 출력
    
    for x in range(len(commands)): # 주어지는 commands만큼의 전체 동작을 위해 선언
        i, j, k = commands[x][0], commands[x][1], commands[x][2] # 변수값을 따로 받아줌
        temp = array[i-1:j] # i 부터 j번 인덱스까지 슬라이싱하여 임시배열에 담아준다.
        temp.sort() # 오름차순 정렬
        answer.append(temp[k-1])
        print(temp)
        
    
    return answer

File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    var answer = [];
    // commands에 주어진 횟수만큼을 반복하며
    for (i = 0; i < commands.length; i++) {
        // 반복시마다 commands내의 범위만큼 잘라낸 임시 배열을 생성해준다
        let tempArr = array.slice(commands[i][0]-1, commands[i][1]);
        // 임시배열을 오름차순으로 정렬 후
        tempArr.sort((a, b) => (a-b))
        // answer 배열에 지정된 자릿수를 담아준다
        answer.push(tempArr[commands[i][2]-1]);
    }
    return answer;
}